File: Android/scenes/showTo1v1/src/main/java/io/agora/scene/showTo1v1/callAPI/extension/CallAPIExtension.kt
package io.agora.scene.showTo1v1.callAPI.extension

import io.agora.scene.showTo1v1.callAPI.CallConfig
import io.agora.scene.showTo1v1.callAPI.PrepareConfig


fun Long.getCostMilliseconds(): Long {
    return System.currentTimeMillis() - this
}

fun PrepareConfig.cloneConfig(): PrepareConfig {
    return PrepareConfig(roomId, rtcToken, rtmToken, localView, remoteView, autoJoinRTC, callTimeoutMillisecond, userExtension)
}

fun CallConfig.cloneConfig(): CallConfig {
    return CallConfig(appId, userId, rtcEngine, rtmClient)
}

File: iOS/AgoraEntScenarios/Common/API/CallAPI/CallAPI/Classes/Extension/CallAPIExtension.swift
//
//  CallAPIExtension.swift
//  CallAPI
//
//  Created by wushengtao on 2023/12/13.
//

import Foundation

extension PrepareConfig {    
    func cloneConfig() -> PrepareConfig {
        let config = PrepareConfig()
        config.roomId = roomId
        config.rtcToken = rtcToken
        config.rtmToken = rtmToken
        config.localView = localView
        config.remoteView = remoteView
//        config.autoAccept = autoAccept
        config.autoJoinRTC = autoJoinRTC
        config.callTimeoutMillisecond = callTimeoutMillisecond
        config.userExtension = userExtension
        return config
    }
}

extension CallConfig {
    func cloneConfig() -> CallConfig {
        let config = CallConfig()
        config.appId = appId
        config.userId = userId
        config.rtcEngine = rtcEngine
        config.rtmClient = rtmClient
        return config
    }
}

extension Date {
    func getCostMilliseconds() -> Int {
        return Int(-timeIntervalSinceNow * 1000)
    }
    
    func millisecondsSince1970() -> Int {
        return Int(round(Date().timeIntervalSince1970 * 1000.0))
    }
}
